src/main/java/org/jabref/gui/preferences/PreviewTabView.java

```diff
@@ -0,0 +1,328 @@
+
+        toRightButton.disableProperty().bind(viewModel.availableSelectionModelProperty().getValue().selectedItemProperty().isNull());
+
+        toLeftButton.disableProperty().bind(viewModel.chosenSelectionModelProperty().getValue().selectedItemProperty().isNull());
+        sortUpButton.disableProperty().bind(viewModel.chosenSelectionModelProperty().getValue().selectedItemProperty().isNull());
+        sortDownButton.disableProperty().bind(viewModel.chosenSelectionModelProperty().getValue().selectedItemProperty().isNull());
+
+        previewPane.setContent(new PreviewViewer(new BibDatabaseContext(), dialogService, Globals.stateManager));
+        // previewPane.setMaxWidth(650); // FixMe: PreviewViewer is too large
+        // previewPane.getContent().maxWidth(650);
+        ((PreviewViewer) previewPane.getContent()).setEntry(TestEntry.getTestEntry());
+        ((PreviewViewer) previewPane.getContent()).setLayout(viewModel.getCurrentLayout());
+        previewPane.visibleProperty().bind(viewModel.chosenSelectionModelProperty().getValue().selectedItemProperty().isNotNull());
+
+        editArea.clear();
+        editArea.setParagraphGraphicFactory(LineNumberFactory.get(editArea));
+        editArea.textProperty().addListener((observable, oldText, newText) -> editArea.setStyleSpans(0, viewModel.computeHighlighting(newText)));
+        editArea.setContextMenu(contextMenu);
+        editArea.visibleProperty().bind(viewModel.chosenSelectionModelProperty().getValue().selectedItemProperty().isNotNull());
+
+        BindingsHelper.bindBidirectional(
+                editArea.textProperty(),
+                viewModel.chosenSelectionModelProperty().getValue().selectedItemProperty(),
+                selectedItem -> update(),
+                text -> {
+                    PreviewLayout selectedItem = viewModel.chosenSelectionModelProperty().getValue().getSelectedItem();
+                    if (selectedItem instanceof TextBasedPreviewLayout) {
+                        ((TextBasedPreviewLayout) selectedItem).setText(editArea.getText().replace("\n", "__NEWLINE__"));
+                    }
+                    update();
+                }
+        );
+
+       // ToDo: Implement selectedIsEditableProperty-Logic in ViewModel
+        readOnlyLabel.visibleProperty().bind(viewModel.selectedIsEditableProperty().not());
+        resetDefaultButton.disableProperty().bind(viewModel.selectedIsEditableProperty().not());
+        contextMenu.getItems().get(0).disableProperty().bind(viewModel.selectedIsEditableProperty().not());
+        contextMenu.getItems().get(2).disableProperty().bind(viewModel.selectedIsEditableProperty().not());
+
+        // editArea.editableProperty().bind(true); // FixMe: Cursor caret disappears
+
+        update();
+    }
+
+    private void update() {
+        PreviewLayout selectedLayout = viewModel.chosenSelectionModelProperty().getValue().getSelectedItem();
```

Comment:
This should ideally all be in the viewModel. If you are in doubt what belongs where: All kid of complex logic, etc should be in the viewModel. 
The View itself should be only some "dummy" just providing the controls for bindings etc and adding menus etc

Reply:
Thanks, this info is really helping me, as I read some tutorials about the mvvm-pattern which were not clear about this.

Reply:
See also the example in our wiki. The other important thing: The viewModel does not have any access to the controls/fx nodes directly. The idea is that the viewModel can be tested independently of the View without having to mock the graphical interface. That's why we always use that bindings stuff. 
https://github.com/JabRef/jabref/wiki/Code---JavaFX